idaes/core/util/env_info.py

```diff
@@ -0,0 +1,130 @@
+
+import idaes
+import pyomo
+import pyomo.environ as pyo
+
+import idaes.ver as ver
+
+class EnvironmentInfo():
+    """Get information about IDAES and the environment IDAES is running in,
+    including OS and Critial dependency versions."""
+
+    known_solvers = [
+        "ipopt",
+        "ipopt_sens",
+        "ipopt_l1",
+        "bonmin",
+        "couenne",
+        "cbc",
+        "k_aug",
+        "dot_sens",
+    ]
+
+    def __init__(self, addtional_solvers=()):
+        # Get idaes version from ver module.  This works even if you just
+        # check a new version our from github, have IDAES installed in-place
+        # and don't reinstall, which is likely mode for a lot of developers
+        self.git_hash = ver.gh
+        self.package_version = ver.package_version
+        self.version_string = ver.__version__
+        self.bin_directory = idaes.bin_directory
+        self.data_directory = idaes.data_directory
+        self.global_config = idaes._global_config_file
+        self.local_config = idaes._local_config_file
+        # Get Python info
+        self.python_version = sys.version
+        # Get os info
+        self.os_platform = platform.system()
+        self.os_release = platform.release()
+        self.os_version = platform.version()
+        # Get pyomo
+        self.pyomo_version = pyomo.version.__version__
+        # Get dependency info
+        self.dependency_versions = OrderedDict()
+        reqs = pkg_resources.get_distribution("idaes-pse").requires()
+        for dep in [x.name for x in reqs]:
+            if dep == "pyomo":
+                continue # pyomo is special
+            try:
+                self.dependency_versions[dep] = \
+                    pkg_resources.get_distribution(dep).version
+            except pkg_resources.DistributionNotFound:
+                self.dependency_versions[dep] = None
+        # Extra packages, users must install these for esoteric features
+        self.extra_versions = OrderedDict()
+        extras = ["seaborn"]
+        for dep in extras:
+            try:
+                self.extra_versions[dep] = \
+                    pkg_resources.get_distribution(dep).version
+            except pkg_resources.DistributionNotFound:
+                self.extra_versions[dep] = None
+        self.solver_versions = {}
+        for s in self.known_solvers + list(addtional_solvers):
+            slv = pyo.SolverFactory(s, validate=False)
+            try:
+                a = slv.available()
+            except AttributeError:
+                a = False
+            if not a:
+                self.solver_versions[s] = None
+            else:
+                v = slv.version()
+                if v is None:
+                    v = "Unknown Version Installed"
+                self.solver_versions[s] = ".".join([f"{x}" for x in v])
+
+
+    def display_dict(self):
+        """Return a dictionary that is in a format that allows easy printing"""
+        d = OrderedDict([
```

Comment:
Minor suggestions for streamlining code: these are completely optional, and I'm mostly adding them for information purposes.

- There's nothing wrong with using an `OrderedDict` to be more explicit, but I think a normal `dict` would be fine since `dict` is guaranteed to keep insertion order since Python 3.7 (this is effectively true also for Python 3.6,  although it's technically implementation-dependent)
- Given the structure of the data (a dict of dicts), defining `d = collections.defaultdict(dict)` (and optionally converting it to a regular dict at the end by having `return dict(d)`) should work and could make the code a bit simpler since you wouldn't have to define the empty keys manually

Reply:
I didn't realized dictionaries were ordered now.  Thanks!  Streamlined things.